src/tools/rbd/action/Nbd.cc

```diff
@@ -121,14 +123,121 @@ int execute_list(const po::variables_map &vm,
   }
 
   return call_nbd_cmd(vm, args, ceph_global_init_args);
+#endif
+}
+
+int execute_attach(const po::variables_map &vm,
+                   const std::vector<std::string> &ceph_global_init_args) {
+#if defined(__FreeBSD__) || defined(_WIN32)
+  std::cerr << "rbd: nbd device is not supported" << std::endl;
+  return -EOPNOTSUPP;
+#else
+  std::vector<std::string> args;
+  std::string device_path;
+
+  args.push_back("attach");
+  std::string img;
+  int r = get_image_or_snap_spec(vm, &img);
+  if (r < 0) {
+    return r;
+  }
+  args.push_back(img);
+
+  if (vm.count("device")) {
+    device_path = vm["device"].as<std::string>();
+    if (!boost::starts_with(device_path, "/dev/nbd")) {
+      std::cerr << "rbd: value of --device should be nbd device name, eg: /dev/nbdX" << std::endl;
+      return -EINVAL;
+    }
```

Comment:
We don't insist that the path starts with "/dev/nbd" on detach and in other places, so I'd drop this check for consistency.